book/src/to_java.md

```diff
@@ -0,0 +1,49 @@
+# The `ToJava` trait
+
+The `ToJava` trait is part of the Duchess prelude. 
+It defines an `&self` method `to_java` that can be used to convert Rust values into Java objects;
+if those Rust types are references to a Java object, then the result is just an identity operation.
+The result of `to_java` is not the Java itself but rather a [`JvmOp`] that produces the Java object.
+
+[`JvmOp`]: ./jvm_op.md
+
+In some cases, the same Rust type can be converted into multiple Java types.
+For example, a Rust Vec can be converted into a Java `ArrayList` but also a Java `List` or `Vector`.
+The `to_java` method takes a type parameter for these cases that can be specified with turbofish,
+e.g., `vec.to_java::<java::util::List<_>>()`.
+
+## Examples
+
+### `String`
+
+The Rust `String` type converts to the Java string type.
+One could compute the Java `hashCode` for a string as follows:
+
+```rust
+let data = format!("Hello, Duchess!");
+let hashCode: i32 =
+    data.to_java()   // Returns a `JvmOp` producing a `java::lang::String`
+        .hashCode()  // Returns a `JvmOp` invoking `hashCode` on this string
```

Comment:
`public int hashCode();` is not declared for `java.lang.String` in `src/java.rs`.

Could you double check this snippet? If I try to compile it (replacing `hashCode` with `length`) I get the following error, but I'm not sure if it's because I miss some import or if it's because the chaining is still unimplemented.
```
error[E0599]: no method named `length` found for struct `ToJavaOp` in the current scope
  --> tests/ui/native_greeting.rs:21:40
   |
21 |     let nameJava: i32 = name.to_java().length().execute();
   |                                        ^^^^^^ method not found in `ToJavaOp<'_, String, String>`
```

Reply:
Yep. We really gotta get the snippets working to the point where they can be auto-compiled.

Reply:
Oh, I see the problem, there's no Deref impl for these ...